### PR TITLE
browzer-wss-underlay

### PR DESCRIPTION
**Pull request by rentallect**

### CODE REVIEW

File: gateway/xgress_edge_wss/dialer.go [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+	}
+
+	if !result.Success {
+		return nil, fmt.Errorf("failed to establish connection with token %v. error: (%v)", token, result.Message)
+	}
+
+	conn := listenConn.newSink(result.NewConnId, dialer.options)
+
+	x := xgress.NewXgress(sessionId, address, conn, xgress.Terminator, &dialer.options.Options)
+	bindHandler.HandleXgressBind(sessionId, address, xgress.Terminator, x)
```
Comment (plorenz) [inline gateway/xgress_edge_wss/dialer.go:99]: You might be out of sync with the latest protocol change. I think there should be an x.Start() here

File: gateway/xgress_edge_wss/listener.go [hunk 2]
```diff
@@ -0,0 +1,450 @@
+/*
```
Comment (plorenz) [inline gateway/xgress_edge_wss/listener.go:1]: I feel like the vast majority of this code could be shared between implementations (I could be wrong, I haven't tried it). I'd be surprised if we couldn't pull at least some of this into a shared package, though

File: gateway/xgress_edge_wss/dialer.go (hunk 1 above)
Comment (rentallect) [reply gateway/xgress_edge_wss/dialer.go:99]: The x.Start() is now in place.

File: gateway/xgress_edge_wss/listener.go (hunk 2 above)
Comment (rentallect) [reply gateway/xgress_edge_wss/listener.go:1]: I agree.  My initial goal was to get things flowing end-to-end (after being away from it for so long), using separate "wss" code, staying away from existing xgress_edge mechanisms.  We can now look into refactoring all xgress_edge_* code using shared code where possible.

File: gateway/xgress_edge_wss/dialer.go [hunk 3]
```diff
@@ -0,0 +1,105 @@
+/*
```
Comment (andrewpmartinez) [inline gateway/xgress_edge_wss/dialer.go:1]: I don't think this xgress component needs to be able to dial. I think this whole thing can be blanked out.

File: gateway/xgress_edge_wss/listener.go (hunk 2 above)
Comment (rentallect) [reply gateway/xgress_edge_wss/listener.go]: deleted.  new wss transport is in place.

File: gateway/xgress_edge_wss/dialer.go (hunk 3 above)
Comment (rentallect) [reply gateway/xgress_edge_wss/dialer.go]: deleted.  new wss transport is in place.

**Review by plorenz — APPROVED** [review]

**Pull request merged**